benches/curve_projective.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2021-2022 Toposware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use rand_core::OsRng;
+
+#[macro_use]
+extern crate criterion;
+
+use criterion::black_box;
+use criterion::Criterion;
+
+extern crate cheetah;
+
+use cheetah::Scalar;
+use cheetah::{AffinePoint, ProjectivePoint};
+
+use cheetah::{BasePointTable, BASEPOINT_TABLE};
+
+static BATCH_SIZES: [u32; 5] = [1, 10, 100, 1000, 10000];
+
+fn criterion_benchmark(c: &mut Criterion) {
+    let mut rng = OsRng;
+    let p = ProjectivePoint::random(&mut rng);
+    let p2 = ProjectivePoint::random(&mut rng);
+    let pow = Scalar::random(&mut rng).to_bytes();
+    let pow2 = Scalar::random(&mut rng).to_bytes();
+
+    let uncompressed_encoding = p.to_uncompressed();
+    let compressed_encoding = p.to_compressed();
+
+    c.bench_function("Projective addition", |bench| {
+        bench.iter(|| black_box(p) + black_box(p2))
+    });
+
+    c.bench_function("Projective mixed addition", |bench| {
+        bench.iter(|| black_box(p) + black_box(p2))
+    });
```

Comment:
What is the difference between these two benches?

Reply:
Oops, you're right p2 is not affine. Will fix it!